packages/menu/src/Menu.ts

```diff
@@ -51,11 +70,61 @@ export class Menu extends SpectrumElement {
      * @private
      */
     public get childRole(): string {
-        return this.getAttribute('role') === 'menu' ? 'menuitem' : 'option';
+        const [selects, role] = this.resolvedSelectsAndRole;
+        if (role === 'menu' || role === 'group') {
+            switch (selects) {
+                case 'single':
+                    return 'menuitemradio';
+                case 'multiple':
+                    return 'menuitemcheckbox';
+                default:
+                    return 'menuitem';
+            }
+        } else {
+            return 'option';
+        }
+    }
+
+    private get resolvedSelects(): undefined | String {
+        return this.resolvedSelectsAndRole[0];
+    }
+
+    private get resolvedSelectsAndRole(): [String | undefined, String | null] {
+        if (this.selects === 'inherit') {
+            let parent = this.parentElement;
```

Comment:
We'd likely want a way to cache this in the future?

Reply:
yes -- I'll give this some thought. Might not be that hard to add.

Reply:
this should be cached now.